Mathlib/CategoryTheory/Abelian/GrothendieckAxioms.lean

```diff
@@ -0,0 +1,259 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Isaac Hernando, Coleton Kotch, Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Adjunction.Limits
+import Mathlib.CategoryTheory.Limits.Filtered
+import Mathlib.CategoryTheory.Limits.FunctorCategory.Basic
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+import Mathlib.CategoryTheory.Limits.Shapes.Biproducts
+
+/-!
+
+# Grothendieck Axioms
+
+This file defines some of the Grothendieck Axioms for abelian categories, and proves
+basic facts about them.
+
+## Definitions
+
+1. `AB4` -- an abelian category satisfies `AB4` provided that coproducts are exact.
+2. `AB5` -- an abelian category satisfies `AB5` provided that filtered colimits are exact.
+
+## Results
+
+- `AB5` implies `AB4`
+
+## Implementation Details
+
+For `AB4` and `AB5`, we only assume left exactness as right exactness is automatic.
+
+## Projects
+
+- Add additional axioms.
+
+-/
+
+namespace CategoryTheory
+
+open Limits Classical
+
+universe v v' u u'
+
+variable (C : Type u) [Category.{v} C]
+
+--Right exactness for arbitrary shapes.
+noncomputable
+example (J : Type u') [Category.{v'} J] [HasColimitsOfShape J C] :
+    PreservesFiniteColimits (colim (J := J) (C := C)) :=
+  inferInstance
```

Comment:
What is the purpose of this example? Removing it allows you to remove the import `Adjunction.Limits` and the `noshake`.

Reply:
Ah, I guess @adamtopaz put it in for teaching purposes, will remove it.